### DISCUSSION TITLE
"autogap" existing hulls

### DISCUSSION THREAD

**Discussion by James-E-A** [FakeFishGames/Barotrauma#9006] [Ideas and suggestions]
When making Hull structures that are too complex for `autohull`, it's a _giant_ pain to Gap them all. But if you don't, the sub ends up [_extremely_ broken](https://steamcommunity.com/sharedfiles/filedetails/?id=2626613976) regarding water flow.

Would it be possible to get `autogap`: _only_ rewrite gaps; connect all adjacent hulls?

**Comment by cshingle** [comment]
I have completed 5 subs, and didn't know autohull or autogap even existed. Adjacent hulls that are connected together with gaps that are typically only used when trying to do complex stuff like follow a curved outer hull.

**Reply by James-E-A** [reply]
> Adjacent hulls that are connected together with gaps that are typically only used when trying to do complex stuff like follow a curved outer hull.

Yes, that is correct. And when I've just carefully made 8 hulls for two external curved surfaces, I don't want to have to manually inject gaps, too, and then also have to remember to change the gaps if I change the hulls at all.

If two hulls are touching on a face, at the very least, a (hypothetical) `autogap` algorithm to detect this should be completely viable, I'd think.
